app/controllers/form_controller.rb

```diff
@@ -1,33 +1,44 @@
 class FormController < ApplicationController
   before_action :authenticate_user!
-  before_action :find_resource, only: [:submit_form]
-  before_action :find_resource_by_named_id, except: [:submit_form]
+  before_action :find_resource, only: [:submit_form, :review]
+  before_action :find_resource_by_named_id, except: [:submit_form, :review]
 
   def submit_form
```

Comment:
Might be worth splitting this into two now: "submit form" and "submit form page" or similar (with their own routes)

Reply:
That would also avoid the need to check referrer I think?